dbos/system_database.py

```diff
@@ -271,12 +278,36 @@ def update_workflow_status(
             )
         else:
             cmd = cmd.on_conflict_do_nothing()
+        cmd = cmd.returning(SystemSchema.workflow_status.c.recovery_attempts)  # type: ignore
 
         if conn is not None:
-            conn.execute(cmd)
+            results = conn.execute(cmd)
```

Comment:
We only provide a `conn` when we flush the result buffer. Therefore, the `if in_recovery` can only be true when we don't have a `conn` object. We can put the entire `if in_recovery` check in the 'else' branch.